@patch('time.sleep')
@patch('subprocess.Popen')
def test_post_rolling_restart_datanode_not_ready(self, process_mock, time_mock):
    process = MagicMock()
    process.communicate.return_value = ['There are no DataNodes here!']
    process.returncode = 0
    process_mock.return_value = process
    try:
        self.executeScript('2.0.6/services/HDFS/package/scripts/datanode.py', classname='DataNode', command='post_rolling_restart', config_file='default.json')
        self.fail('Missing DataNode should have caused a failure')
    except Fail as fail:
        self.assertTrue(process_mock.called)
        self.assertEqual(process_mock.call_count, 12)
